substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/graal/code/SubstrateBackend.java

```diff
@@ -152,14 +164,83 @@ public CompilationResult newCompilationResult(CompilationIdentifier compilationI
         return new CompilationResult(compilationIdentifier, name) {
             @Override
             public void close(OptionValues options) {
-                /*
-                 * Do nothing, we do not want our CompilationResult to be closed because we
-                 * aggregate all data items and machine code in the native image heap.
-                 */
+                FrameOptimizer.compressInfoPoints(this);
```

Comment:
I wonder if it's worth skipping this altogether when debug info generation is disabled.

Reply:
Perhaps. However, since there will be very few infopoints it will not cost much. What I must do is update the comment.